### DISCUSSION TITLE
Receiving multiple 'action' messages from a single button press on Ikea E2002

### DISCUSSION THREAD

**Discussion by NCopeman** [Koenkk/zigbee2mqtt#10014] [Devices]
Hi, 
I've just received my Sonoff Zigbee dongle and I'm pairing my first devices - an Ikea Strybar remote (E2002), and a Tradfri dimmer (E1743). I'm intending on using ControllerX to map double/triple presses to different functions. This is working fine with the E1743, but with the E2002 if I have any double-press actions mapped to the remote both single & double-presses cease functioning. I noticed that in the Z2M logs I see multiple payloads published to the 'zigbee2mqtt/Bedroom Light Switch/action' topic for a single button press on the E2002, but only a single message published for a single press on the E1743. 

The E1743 (working):
```
Info MQTT publish: topic 'zigbee2mqtt/Bedroom Dimmer', payload '{"action":"on","battery":87,"click":null,"linkquality":90,"update_available":null}'
Info MQTT publish: topic 'zigbee2mqtt/Bedroom Dimmer', payload '{"action":"","battery":87,"click":null,"linkquality":90,"update_available":null}'
Info MQTT publish: topic 'zigbee2mqtt/Bedroom Dimmer/action', payload 'on'
```

The E2002 (not working):
```
Info MQTT publish: topic 'zigbee2mqtt/Bedroom Light Switch', payload '{"action":"on","battery":85,"linkquality":45,"update":{"state":"idle"},"update_available":false}'
Info MQTT publish: topic 'zigbee2mqtt/Bedroom Light Switch', payload '{"action":"","battery":85,"linkquality":45,"update":{"state":"idle"},"update_available":false}'
Info MQTT publish: topic 'zigbee2mqtt/Bedroom Light Switch/action', payload 'on'
Info MQTT publish: topic 'zigbee2mqtt/Bedroom Light Switch', payload '{"action":"on","action_group":901,"battery":85,"linkquality":45,"update":{"state":"idle"},"update_available":false}'
Info MQTT publish: topic 'zigbee2mqtt/Bedroom Light Switch', payload '{"action":"","battery":85,"linkquality":45,"update":{"state":"idle"},"update_available":false}'
Info MQTT publish: topic 'zigbee2mqtt/Bedroom Light Switch/action', payload 'on'
Info MQTT publish: topic 'zigbee2mqtt/Bedroom Light Switch', payload '{"action":"on","battery":85,"linkquality":45,"update":{"state":"idle"},"update_available":false}'
Info MQTT publish: topic 'zigbee2mqtt/Bedroom Light Switch', payload '{"action":"","battery":85,"linkquality":45,"update":{"state":"idle"},"update_available":false}'
Info MQTT publish: topic 'zigbee2mqtt/Bedroom Light Switch/action', payload 'on'
```
(note the line` Info MQTT publish: topic 'zigbee2mqtt/Bedroom Light Switch/action', payload 'on'` is repeated three times for a single button press).
Is this expected behaviour? Both devices are listed on the support list.

**Comment by bartplessers** [comment]
Solved in latest dev release, see #10089

**Reply by NCopeman** [reply]
Thanks for the heads-up!

**Comment by NCopeman** [comment]
Can confirm the issue's fixed as of release 1.22.1

**Reply by jmgutierrezm** [reply]
Hi, I have the zigbee2mqtt v1.25.0-1 installed and problem remains for IKEA E2002. I always get three lines for each single button press:

Info MQTT publish: topic 'zigbee2mqtt/Mando interior', payload '{"action":"on","battery":100,"linkquality":65,"update":{"state":"idle"},"update_available":null}'
Info MQTT publish: topic 'zigbee2mqtt/Mando interior', payload '{"action":"","battery":100,"linkquality":65,"update":{"state":"idle"},"update_available":null}'
Info MQTT publish: topic 'zigbee2mqtt/Mando interior/action', payload 'on'

**Comment by VladimirV01** [comment]
Hello @Koenkk,

Not sure if it is ok to comment in such an old post, please let me know if this is not the correct way to do things.

I wanted to report that I have this same issue in the latest 1.40.0 release.
By "the same issue" I mean the fact that the action is published 2 times.
First as <b>JSON</b> in topic <b>zigbee2mqtt/Multiswitch</b> and the second time as <b>clear text</b> in topic <b>zigbee2mqtt/Multiswitch/action</b>.

```
info 2024-09-04 22:26:23z2m:mqtt: MQTT publish: topic 'zigbee2mqtt/Multiswitch', payload '{"action":"2_single","battery":null,"linkquality":61}'
info 2024-09-04 22:26:23z2m:mqtt: MQTT publish: topic 'zigbee2mqtt/Multiswitch', payload '{"action":"","battery":null,"linkquality":61}'
info 2024-09-04 22:26:23z2m:mqtt: MQTT publish: topic 'zigbee2mqtt/Multiswitch/action', payload '2_single'
info 2024-09-04 22:26:30z2m:mqtt: MQTT publish: topic 'zigbee2mqtt/Multiswitch', payload '{"action":"2_hold","battery":null,"linkquality":65}'
info 2024-09-04 22:26:30z2m:mqtt: MQTT publish: topic 'zigbee2mqtt/Multiswitch', payload '{"action":"","battery":null,"linkquality":65}'
info 2024-09-04 22:26:30z2m:mqtt: MQTT publish: topic 'zigbee2mqtt/Multiswitch/action', payload '2_hold'
```

Can I fix it somehow?

**Reply by Koenkk** [reply]
this is because you have the following in your configuration.yaml:

```yaml
advanced:
  output: attribute_and_json
```

(see https://www.zigbee2mqtt.io/guide/configuration/mqtt.html#mqtt-behaviour for more info)

**Reply by VladimirV01** [reply]
I did google this issue before posting my comment here, but now I realize that I didn't google it hard enough. 

Sorry for bothering and thank you very much for pointing it out to me and for your great work on this project!
